lib/libc/minimal/source/stdlib/strtoul.c

```diff
@@ -99,7 +99,7 @@ unsigned long strtoul(const char *nptr, char **endptr, register int base)
 	if (any < 0) {
 		acc = ULONG_MAX;
 		errno = ERANGE;
-	} else if (neg) {
+	} else if (neg != 0) {
```

Comment:
What about the rest in this function?

For instance:
```
L106: *endptr = (char *)(any ? s - 1 : nptr);
```

Reply:
The line you mention do violate a different MISRA C:2012 guideline. It will be amended in a future commit and PR.